docs/DeveloperGuide.adoc

```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+*API* : link:{repoURL}/src/main/java/seedu/moneygowhere/storage/Storage.java[`Storage.java`]
 
 The `Storage` component,
 
 * can save `UserPref` objects in json format and read it back.
-* can save the Address Book data in json format and read it back.
+* can save the Spending Book data in json format and read it back.
 
 [[Design-Commons]]
 === Common classes
 
-Classes used by multiple components are in the `seedu.addressbook.commons` package.
+Classes used by multiple components are in the `seedu.moneygowhere.commons` package.
 
 == Implementation
 
 This section describes some noteworthy details on how certain features are implemented.
 
+=== Statistics feature
+==== Current Implementation
+For the current `statistics` feature, there are 2 main commands that the user can execute.
+
+1. `stats` - display statistics for all spending
+2. `stats d/DATE_START d/DATE_END` - display statistics for spending within a date range
+
+The `StatsCommandParser` differentiates these 2 commands based on whether a valid date range is provided as illustrated in the diagram below.
+
+.Activity Diagram showing behaviour of StatsCommandParser
+image::statsactivitydiagram.png[width="600"]
+
+The implementation of the Statistics feature can be split into 2 phases, preparation and execution. Given below is an explanation of how the Statistics mechanism behaves at each phase.
+
+===== Preparation
+
+During the preparation phase, the program parses the command for Statistics and prepares to open a new Statistics Window.
+The following sequence diagram shows how `stats d/01/01/2019 d/07/01/2019` works during the preparation phase.
+
+.Sequence Diagram demonstrating the preparation stage of Statistics feature
+image::StatsSequenceDiagram.png[]
+
+1. User executes `stats d/01/01/2019 d/07/01/2019` to generate statistics for spending within `01/01/2019` and `07/01/2019`.
+
+2. SpendingBookParser will parse and identify the command as a StatsCommand and pass on the argument `d/ 01/01/2019 d/07/01/2019` to StatsCommandParser.
+
+3. StatsCommandParser will check whether the date range parameters provided are valid.
+
+4. `StatsCommand#execute()` is executed, returning a message which tells the user that the Stats Window will be opened and set a boolean `isShowStats` = true to indicate that a new window to display the statistics will be opened.
+
+===== Execution
+
+In the execution phase, the program processes and retrieves the data to be displayed and finally displaying it.
+
+There are 2 types of data that the program require, `statsMessage` for displaying the statistics in text and `statsData` for displaying the data in a piechart.
+Below is the UML sequence diagram and a step-by-step explanation for the retrieval of `statsData`. `statsMessage` is retrieved in the same manner.
+
+.Sequence Diagram of Execution Phase showing retrieving of `statsData`
+image::StatsMessageSequenceDiagram.png[]
```

Comment:
Should True in the call be `true` with the lowercase instead?

Should the lifeline of StatsCommandParser continue after the cross?

Should Model have an activation box?

Should the return arrows be dotted lines instead?

Is CommandResult meant to be there? It seems to have no interaction.